Lib/sqlite3/test/types.py

```diff
@@ -255,7 +256,15 @@ def test_number2(self):
     def test_convert_zero_sized_blob(self):
         self.con.execute("insert into test(cbin) values (?)", (b"",))
         cur = self.con.execute("select cbin from test")
-        self.assertEqual(cur.fetchone()[0], b"blobish")
+        # Zero-sized blobs with converters returns None.  This differs from
+        # blobs without a converter, where b"" is returned.
+        self.assertEqual(cur.fetchone()[0], None)
+
+    def test_convert_null_date(self):
+        cur = self.con.execute("insert into test(d) values (null)")
+        cur.execute("select d from test")
+        values = [x[0] for x in cur.fetchall()]
+        self.assertEqual(values, [None])
```

Comment:
Actually, correcting `test_convert_zero_sized_blob` should be enough; `test_convert_null_date` does not add anything new to the test suite. @berkerpeksag?

Reply:
Yes. But it would be nice to make sure that there is a test for the case where we check zero-sized blobs without a converter returns `b''`.

Reply:
We already have that: https://github.com/python/cpython/blob/37d519b86e761946831747b61c6390086b17ecc8/Lib/sqlite3/test/regression.py#L412-L415

Reply:
Then removing `test_convert_null_date` seems to be the only blocker here.

Reply:
Fixed in 5f49a912423555c43078361dfa04e42a6dba3d31